ac_dc/oscar_sample_filter.py

```diff
@@ -403,13 +403,26 @@ def __init__(
         path_kenlm_model,
         num_proc,
         path_dir_save_oscar,
+        use_registry,
+        registry_data_files,
     ):
         self.lang_oscar_id = lang_oscar_id
         self.path_fasttext_model = path_fasttext_model
         self.path_kenlm_model = path_kenlm_model
-        self.ds = load_dataset(
-            "oscar", f"unshuffled_deduplicated_{self.lang_oscar_id}"
-        )["train"]
+        if use_registry:
+            data_files = (
+                registry_data_files
+                if registry_data_files
+                else f"{lang_oscar_id}/*.jsonl.gz"
+            )
+            self.ds = load_dataset(
+                "mhtoin/register_oscar",
+                data_files=data_files,
+            )["train"]
+        else:
+            self.ds = load_dataset(
```

Comment:
@HugoLaurencon eventually, we'll want to run the pipeline on other datasets (mC4),  not just Oscar. We will need to decouple the dataset object from the filtering to that. If so, we can ignore my changes and make the dataset loading more flexible

Reply:
Closing this PR, as I think it just makes more sense to load any dataset from the main class